deploy/docker/hail_search/start_server.sh

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+gunicorn hail_search.web_app:init_web_app -c /gunicorn_config.py
```

Comment:
I opted for `gunicorn` here for consistency with the Django app and because it's more "best-practicey".  I'm not sure it's necessary though, and it requires a pip install inside the docker container.

[Docs here](https://docs.aiohttp.org/en/stable/deployment.html#start-gunicorn)

Reply:
I don't love it, and I don't really understand what advantage we are getting by having it. Aiohttp seems to work just fine on its own

Reply:
yeah, I don't love it either.

I think my gut was to go for it because `seqr` is using it, and I've consistently seen the `nginx -> gunicorn -> python web framework` stack, with `gunicorn` doing the process management (how many workers, what happens when they die etc, routing concurrent requests to different workers) as well as the gateway interface stuff.  I _think_ the gateway stuff is unnecessary because aiohttp has support for it (in ways that Django and Flask do not).

Regarding concurrency, I _think_ we only ever want to run a single worker here right?  So that a single request blocks every subsequent request while it's running?  With concurrency being managed at the k8s level (I want 4 containers running, with load distributed by the nginx-ingress)?  In that case, we should be fine to remove `gunicorn`.  

@sjahl if you had any thoughts.

Reply:
historically, i've always had gunicorn on python web app deployments for the reasons Ben mentioned... the app servers from flask and django are usually not suitable for production without gunicorn and a webserver in front of them.

I can see the process lifecycle stuff being useful here, for restarting processes that crash? if the app containers spawn quickly and kubernetes can determine that dead containers need replacing, maybe it's less critical. But, we'd also want to keep in mind that the k8s health check would need to be fairly conservative to avoid killing pods unnecessarily... so you're probably looking at a several second health check interval, and a certain number of consecutive failures before taking action.

In any case, if this is only going to dev, i'd be pro experimentation and seeing how well it handles things without the gunicorn layer.

Reply:
+1

I think I'm generally in favor of putting the concurrency management in the kubernetes here and only having a single process, that if it fails causes the Docker container to exit, run as a server.  This means that our health checks will be completely inaccessible while a hail query or any other request is running, but allows us to prevent multiple hail queries running concurrently within the same container.  Our health check will need to be very very conservative.

Reply:
I found [a question](https://github.com/aio-libs/aiohttp/discussions/6805) which is unanswered, but at least points to somebody else having similar thoughts.

Reply:
And a [decent discussion](https://github.com/aio-libs/aiohttp/issues/552) about deployment